plugins/modules/route53.py

```diff
@@ -567,8 +622,8 @@ def main():
     if command_in == 'create' or command_in == 'delete':
         if alias_in and len(value_in) != 1:
             module.fail_json(msg="parameter 'value' must contain a single dns name for alias records")
-        if (weight_in is None and region_in is None and failover_in is None) and identifier_in is not None:
-            module.fail_json(msg="You have specified identifier which makes sense only if you specify one of: weight, region or failover.")
+        if (weight_in is None and region_in is None and failover_in is None and geo_location is None) and identifier_in is not None:
```

Comment:
You can also probably rewrite like this:
```suggestion
        if not all([weight_in, region_in, failover_in, geo_location]) and identifier_in is not None:
```

Reply:
Fixed.